### ISSUE TITLE
dbdeployer fails to autodetect pxc flavor from binaries of 5.6.x versions

### ISSUE THREAD

**Issue by arronax** [datacharmer/dbdeployer#118]
**Describe the bug**
dbdeployer fails to autodetect pxc flavor from binaries of 5.6.x versions. You have to force the flavor using `--flavor=pxc`.

**To Reproduce**
1. Get any generic binaries for PXC 5.6 from https://www.percona.com/downloads/Percona-XtraDB-Cluster-56/
2. Run, for example:
```
$ ~/bin/dbdeployer --sandbox-binary /opt/percona_xtradb_cluster deploy --topology=pxc replication 5.6.43 --base-port=25643
```
3. dbdeployer refuses to deploy the cluster:
```
error creating sandbox: ''Xtradb Cluster' requires flavor 'pxc' version '5.6.14'+'
```

**Expected behavior**
dbdeployer should be able to automatically detect pxc flavor from 5.6 binaries as it does for 5.7 and 8.0

**Environment:**
 - OS: CentOS 7.8
 - dbdeployer version 1.54.0
 - tarball Percona-XtraDB-Cluster-5.6.43-rel84.3-28.32.1.Linux.x86_64.ssl102.tar.gz
 - tarball origin https://www.percona.com/downloads/Percona-XtraDB-Cluster-56/

**Additional context**
The problem is that 5.6 tarball doesn't contain files expected for pxc flavor in FlavorCompositionList. Seven different files are checked there, only two of them are present in 5.6 tarball. For whatever reason, 5.6 tarballs do not include libperconaserverclient. Instead they ship with libmysqlclient.
```
$ find 5.6.* -type f \( -name "garbd" -o -name "libgalera_smm.a" -o -name "libgalera_smm.dylib" -o -name "libgalera_smm.so" -o -name "libperconaserverclient.a" -o -name "libperconaserverclient.dylib" -o -name "libperconaserverclient.so" \)
5.6.22/bin/garbd
5.6.22/lib/libgalera_smm.so
5.6.24/bin/garbd
5.6.24/lib/libgalera_smm.so
5.6.35/bin/garbd
5.6.35/lib/libgalera_smm.so
5.6.36/bin/garbd
5.6.36/lib/libgalera_smm.so
5.6.38/bin/garbd
5.6.38/lib/libgalera_smm.so
5.6.43/bin/garbd
5.6.43/lib/libgalera_smm.so
5.6.49/bin/garbd
5.6.49/lib/libgalera_smm.so
```
Compared with 5.7 or 8.0
```
$ find 5.7.* -type f \( -name "garbd" -o -name "libgalera_smm.a" -o -name "libgalera_smm.dylib" -o -name "libgalera_smm.so" -o -name "libperconaserverclient.a" -o -name "libperconaserverclient.dylib" -o -name "libperconaserverclient.so" \)
5.7.14/bin/garbd
5.7.14/lib/libgalera_smm.so
5.7.14/lib/libperconaserverclient.a
...
5.7.31/bin/garbd
5.7.31/lib/libgalera_smm.so
5.7.31/lib/libperconaserverclient.a

$ find 8.0.* -type f \( -name "garbd" -o -name "libgalera_smm.a" -o -name "libgalera_smm.dylib" -o -name "libgalera_smm.so" -o -name "libperconaserverclient.a" -o -name "libperconaserverclient.dylib" -o -name "libperconaserverclient.so" \)
8.0.18/bin/garbd
8.0.18/lib/libgalera_smm.so
8.0.18/lib/libperconaserverclient.a
8.0.19/bin/garbd
8.0.19/lib/libgalera_smm.so
8.0.19/lib/libperconaserverclient.a
```

**Comment by datacharmer** [comment]
Thanks for this detailed report.

I have a few questions:

1. how did you create the directory `/opt/percona_xtradb_cluster` ? did you unpack manually or use `dbdeployer unpack`?
2. Is there a particular reason for using such old versions of PXC? Anything that makes it convenient over the more recent ones?

**Comment by arronax** [comment]
1. In this particular case, and for this directory, it was a manual unpacked version. I see that `dbdeployer unpack` puts the `FLAVOR` file into the unpacked directory, and "faking" that file is a yet another way to fix the issue.
2. I have work-related reasons to test stuff now and then with PXC 5.6. I understand, of course, that it's extremely old. Wouldn't recommend anyone to use it.

I put together a small fix in #119, the particular addition to `FlavorCompositionList` is unique and probably shouldn't overlap with other possible types of binaries. A more robust way would be to look for _libmysqlclient.so.18.1.0_, but it looks like an overkill to add for PXC 5.6, which will go away, eventually.

**Comment by datacharmer** [comment]
Thanks also for the fix. A preliminary test suite run confirms that it works. I will release the new binaries soon

**Comment by datacharmer** [comment]
Fixed in [release 1.54.1](https://github.com/datacharmer/dbdeployer/releases/tag/v1.54.1)
